src/app/courses/components/add-course/add-course.component.ts

```diff
@@ -1,9 +1,25 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, Input, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material';
+import { Course } from '@app/shared/models/course.model';
 
 @Component({
   selector: 'app-add-course',
   templateUrl: './add-course.component.html',
   styleUrls: ['./add-course.component.scss']
 })
 export class AddCourseComponent {
+  course: Course = null;
+
+  constructor(public dialogRef: MatDialogRef<AddCourseComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: any) {
+    this.course = data.course;
+   }
+
+  save() {
```

Comment:
Please don't leave this method empty. You may close your dialog ( and print into console your course ), for example.